src/devices/Pn5180/Pn5180.cs

```diff
@@ -0,0 +1,1496 @@
+        }
+
+        /// <summary>
+        /// Is the external field activated?
+        /// </summary>
+        /// <returns>True if active, false if not</returns>
+        public bool IsRadioFrequenceFieldExternal()
+        {
+            Span<byte> status = stackalloc byte[4];
+            try
+            {
+                SpiReadRegister(Register.RF_STATUS, status);
+            }
+            catch (TimeoutException)
+            {
+                return false;
+            }
+
+            return (status[2] & 0b0000_0100) == 0b0000_0100;
+        }
+
+        private bool SetRadioFrequence(bool field)
+        {
+            Span<byte> rfConfig = stackalloc byte[2];
+            rfConfig[0] = (byte)(field ? Command.RF_ON : Command.RF_OFF);
+            rfConfig[1] = (byte)RadioFrequenceCollision;
+            try
+            {
+                SpiWrite(rfConfig);
+            }
+            catch (Exception)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        #endregion
+
+        #region Listen to cards
+
+        /// <summary>
+        /// Listen to any 14443 Type A card
+        /// </summary>
+        /// <param name="transmitter">The transmitter configuration, should be compatible with type A card</param>
+        /// <param name="receiver">The receiver configuration, should be compatible with type A card</param>
+        /// <param name="card">The type A card once detected</param>
+        /// <param name="timeoutPollingMilliseconds">The time to poll the card in milliseconds. Card detection will stop once the detection time will be over</param>
+        /// <returns>True if a 14443 Type A card has been detected</returns>
+        public bool ListenToCardIso14443TypeA(TransmitterRadioFrequenceConfiguration transmitter, ReceiverRadioFrequenceConfiguration receiver, out Data106kbpsTypeA card, int timeoutPollingMilliseconds)
+        {
+            card = new Data106kbpsTypeA();
+            // From NXP documentation AN12650.pdf, Page 8 and forward
+            // From NXP documentation AN10833.pdf page 10
+            // From TI documentation http://www.ti.com/lit/an/sloa136/sloa136.pdf page 7 and 6 for the flow of selecting
+            // and getting the UID
+            // Load the configuration for the specific card
+            var ret = LoadRadioFrequenceConfiguration(transmitter, receiver);
+            // Switch on the radio frequence field
+            ret &= SetRadioFrequence(true);
+
+            Span<byte> atqa = stackalloc byte[2];
+            Span<byte> irqStatus = stackalloc byte[4];
+            Span<byte> uidSak = stackalloc byte[7];
+            Span<byte> uid = stackalloc byte[10];
+            Span<byte> sak = stackalloc byte[1];
+            Span<byte> sakInterm = stackalloc byte[5];
+            int numBytes;
+
+            DateTime dtTimeout = DateTime.Now.AddMilliseconds(timeoutPollingMilliseconds);
```

Comment:
this code here (this method) looks super similar to PN532, is it possible to unify them both?

Reply:
I made this one as similar as possible as the PN532, keeping the philosophy. Issue with the PN532 is that it's working in a different way as the PN532 returns a byte array that then you have to try to understand what you get as a result. 
But, whenever I'll have a fix to do to the PN532, I can add *similar* functions to the PN532 to listen to specific types of cards. That said, parameters will be a bit different as the PN532 dos not have this concept of receiver and transmitter frequencies settings, it has only 1 for both and the choices as more limited on the PN532. So It won't be the same settings.
But clearly, trying to have a similar approach and similar naming to make it easier to use.